### PR TITLE
Fixing tests for interop and adding regex,predicate tests. #fix 763

### PR DESCRIPTION
**Pull request by msft-Jeyaram**

### CODE REVIEW

File: tests/unittests/Foundation/NSRegularExpressionTests.m [hunk 1]
```diff
@@ -111,6 +111,20 @@
     ASSERT_OBJCEQ(@"Windows Objective C bridge is amazing", outputString);
 }
 
+NSUInteger checkMatches(NSRegularExpression* regex, NSString* testString) {
+    NSRange range = { 0, [testString length] };
+    return [regex numberOfMatchesInString:testString options:0 range:range];
+}
+
+TEST(NSRegularExpression, MatchtingTest0) {
```
Comment (ms-jihua) [inline tests/unittests/Foundation/NSRegularExpressionTests.m]: >t [](start = 31, length = 1)

typo

Comment (msft-Jeyaram) [inline tests/unittests/Foundation/NSRegularExpressionTests.m]: ops :dragon:

File: tests/unittests/Foundation/NSPredicateTests.m [hunk 2]
```diff
@@ -310,9 +306,9 @@ - (void)dealloc {
 }
 
 TEST(NSPredicate, Matches) {
-    NSMutableArray* animals = [[[NSMutableArray alloc] init] autorelease];
+    NSMutableArray* animals = [NSMutableArray array];
 
-    PredicateTestAnimal* animal = [[[PredicateTestAnimal alloc] initWithName:@"test1" age:5 speed:115] autorelease];
+    PredicateTestAnimal* animal = [[[PredicateTestAnimal alloc] initWithName:@"test" age:5 speed:115] autorelease];
```
Comment (ms-jihua) [inline tests/unittests/Foundation/NSPredicateTests.m]: what's the context behind this change?

File: tests/unittests/Foundation/NSPredicateTests.m [hunk 3]
```diff
@@ -113,20 +112,17 @@ - (instancetype)initWithValue:(NSInteger)value {
 
     NSPredicate* unArchivedPredicate = [NSKeyedUnarchiver unarchiveObjectWithData:data];
     ASSERT_TRUE_MSG(unArchivedPredicate != nil, "FAILED: unArchivedPredicate should be non-null!");
-
-    [predicate release];
 }
 
 TEST(NSPredicate, copy) {
-    NSPredicate* predicate = [[NSPredicate alloc] init];
+    NSPredicate* predicate = [NSPredicate predicateWithFormat:@"1<3"];
```
Comment (DHowett-MSFT) [inline tests/unittests/Foundation/NSPredicateTests.m]: >1<3"] [](start = 64, length = 5)

Is `TRUE` a valid predicate? If so, it might be more straightforward to just use that.

File: tests/unittests/Foundation/NSPredicateTests.m [hunk 4]
```diff
@@ -113,20 +112,17 @@ - (instancetype)initWithValue:(NSInteger)value {
...
     ASSERT_TRUE_MSG(predicate != nil, "FAILED: predicate should be non-null!");
 
     NSPredicate* copyObj = [predicate copy];
     ASSERT_TRUE_MSG(copyObj != nil, "FAILED: copyObj should be non-null!");
 
     ASSERT_OBJCEQ_MSG(predicate, copyObj, "FAILED: objects do not match.");
 
-    [copyObj release];
```
Comment (DHowett-MSFT) [inline tests/unittests/Foundation/NSPredicateTests.m]: >[copyObj release]; [](start = 4, length = 18)

Why did you remove this release? The copy is still +1.

**Comment by DHowett-MSFT** [comment]
:clock1:

File: tests/unittests/Foundation/NSPredicateTests.m (hunk 4 above)
Comment (rajsesh-msft) [inline tests/unittests/Foundation/NSPredicateTests.m]: instead should remove [predicate release];

---
In reply to: [73250756](https://github.com/Microsoft/WinObjC/pull/788#discussion_r73250756) [](ancestors = 73250756)

**Comment by rajsesh-msft** [comment]
:clock1:

File: tests/unittests/Foundation/NSPredicateTests.m (hunk 4 above)
Comment (msft-Jeyaram) [inline tests/unittests/Foundation/NSPredicateTests.m]: accidental, it was supposed to be the other.

---
In reply to: [73251344](https://github.com/Microsoft/WinObjC/pull/788#discussion_r73251344) [](ancestors = 73251344,73250756)

File: tests/unittests/Foundation/NSPredicateTests.m (hunk 3 above)
Comment (msft-Jeyaram) [inline tests/unittests/Foundation/NSPredicateTests.m]: Given this is just to obtain a valid object, this should be fine for the scope of this test.

---
In reply to: [73250728](https://github.com/Microsoft/WinObjC/pull/788#discussion_r73250728) [](ancestors = 73250728)

File: tests/unittests/Foundation/NSPredicateTests.m (hunk 2 above)
Comment (msft-Jeyaram) [inline tests/unittests/Foundation/NSPredicateTests.m]: this is due to the fact that with created options, when you do the regex match you need an absolute match. We weren't supporting the full array of options, since we didn't have the functionality to map them.
Now there is coreFoundation. I'm going to send out the changes in that CR (for few of the old todos, that can be supported now).
 

---
In reply to: [73245208](https://github.com/Microsoft/WinObjC/pull/788#discussion_r73245208) [](ancestors = 73245208)

File: tests/unittests/Foundation/NSPredicateTests.m [hunk 5]
```diff
@@ -324,8 +320,26 @@ - (void)dealloc {
     ASSERT_EQ_MSG(1, [validTestAnimals count], "FAILED: Unable to filter based on class property.");
 }
 
-TEST(NSPredicate, Like) {
-    NSMutableArray* animals = [[[NSMutableArray alloc] init] autorelease];
+TEST(NSPredicate, Matches0) {
```
Comment (DHowett-MSFT) [inline tests/unittests/Foundation/NSPredicateTests.m]: Is it possible to get better names for these tests, since you're touching them?
`NSPredicate, StringsFilteredByRegularExpression`
`NSPredicate, InstanceFieldsFilteredByRegularExpression`

File: tests/unittests/Foundation/NSRegularExpressionTests.m [hunk 6]
```diff
@@ -111,6 +111,20 @@
     ASSERT_OBJCEQ(@"Windows Objective C bridge is amazing", outputString);
 }
 
+NSUInteger checkMatches(NSRegularExpression* regex, NSString* testString) {
+    NSRange range = { 0, [testString length] };
+    return [regex numberOfMatchesInString:testString options:0 range:range];
+}
+
+TEST(NSRegularExpression, MatchingTests0) {
```
Comment (DHowett-MSFT) [inline tests/unittests/Foundation/NSRegularExpressionTests.m]: Is there a better name for this test?

**Comment by msft-Jeyaram** [comment]
In CI Build

**Comment by msft-Jeyaram** [comment]
ping!

File: tests/unittests/Foundation/NSPredicateTests.m [hunk 7]
```diff
@@ -17,10 +17,9 @@
 #import <TestFramework.h>
 #import <Foundation/Foundation.h>
 
-TEST(NSPredicate, Init) {
-    NSPredicate* predicate = [[NSPredicate alloc] init];
+TEST(NSPredicate, predicate) {
```
Comment (ms-jihua) [inline tests/unittests/Foundation/NSPredicateTests.m]: >p [](start = 18, length = 1)

nit: pref uppercase

**Comment by ms-jihua** [comment]
:shipit:

**Comment by msft-Jeyaram** [comment]
In CI Build

**Comment by msft-Jeyaram** [comment]
The Dispatch.autorelease test was hit again. As per previous discussion, i'll go ahead with the merge.
